tools/serverpod_cli/test/database/extentions/migration_with_many_to_many_relation_pgsql_code_generation_test.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/database/create_definition.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:serverpod_service_client/serverpod_service_client.dart';
+import 'package:test/test.dart';
+
+import '../../generator/dart/client_code_generator/class_constructors_test.dart';
+
+void main() {
+  test(
+      'Given a table that is not managed by serverpod that changes to be managed',
+      () {
+    DatabaseDefinition sourceDefinition;
+    {
+      var models = [
```

Comment:
Started from the high-level model sources here, because the set-up is rather complex and I would not trust (myself) getting the low-level classes setup exactly in the way they would be from this definition.

So a more higher level (integration-ish) test makes sense for me, as that's also what the user sees.